src/main/java/io/appform/dropwizard/sharding/dao/RelationalDao.java

```diff
@@ -71,13 +75,33 @@ public RelationalDaoPriv(SessionFactory sessionFactory) {
             this.sessionFactory = sessionFactory;
         }
 
-        T get(Object lookupKey) {
-            return uniqueResult(currentSession()
-                    .createCriteria(entityClass)
-                    .add(Restrictions.eq(keyField.getName(), lookupKey))
-                    .setLockMode(LockMode.READ));
+
+        /**
+         * Get a row matching the field annotated with {@code @Id} annotation
+         *
+         * @param lookupKey ID for which data needs to be fetched
+         */
+
+        T get(final Object lookupKey) {
+            val q = createQuery(currentSession(),
+                    entityClass,
+                    (queryRoot, query, criteriaBuilder) ->
+                            query.where(criteriaBuilder.equal(queryRoot.get(keyField.getName()), lookupKey)));
+            return uniqueResult(q.setLockMode(LockModeType.NONE));
```

Comment:
TODO: This is a change from current behaviour. Needs to be discussed with @santanusinha

Reply:
Removing this for now. Will raise this in a separate MR